build_and_deploy/orchestrator.ts

```diff
@@ -40,7 +47,22 @@ export class Orchestrator {
             let artifactsToDeploy: Resource[][] = await getArtifacts(armParameterContent, armTemplateContent, overrideArmParameters,
                 this.targetWorkspace, targetLocation);
 
+            if(this.deleteArtifactsNotInTemplate)
+            {
+                // Delete extra artifacts in the workspace
+                SystemLogger.info("Attempting to delete artifacts from workspace, that were not in the template.");
+                var artifactsInWorkspace = await getArtifactsFromWorkspace(this.targetWorkspace, this.environment);
+                SystemLogger.info(`Found ${artifactsInWorkspace.length} artifacts in the workspace.`);
+                var artifactsToDeleteInWorkspace = getArtifactsToDeleteFromWorkspace(artifactsInWorkspace, artifactsToDeploy, typeMap);
+                SystemLogger.info(`Found ${artifactsToDeleteInWorkspace.length} artifacts in the workspace that many need to be deleted.`);
+                var artifactsToDeleteInWorkspaceInOrder = getArtifactsToDeleteFromWorkspaceInOrder(artifactsToDeleteInWorkspace);
+                await this.deleteResourcesInOrder(this.artifactClient, artifactsToDeleteInWorkspaceInOrder!, this.targetWorkspace, this.environment, armParameterContent);
+                SystemLogger.info("Completed deleting artifacts from workspace, that were not in the template.");
```

Comment:
any exception handling needed here? incase of deleting artifacts failed due to what ever reason

Reply:
This is already inside a try block